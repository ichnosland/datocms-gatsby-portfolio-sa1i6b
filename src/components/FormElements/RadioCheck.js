/**
*
* RadioCheck
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import { colore } from 'style/color';
import media from 'style/mediainjector';

export const RadioCheckLabel = styled.label`
  position: relative;
  display: block;
  width: 100%;
  margin: 10px 0;
  background-color: ${(props) => props.bgcolor ? props.bgcolor : colore.ui.contrast};
  ${media.desktop`
    border: 1px solid rgba(0, 0, 0, 0.05);
    border-bottom-width: 2px;
  `}
  ${({ type, hasImage }) => {
    if (!hasImage && type === 'radio') {
      return css`
        border-radius: 100px;
        padding: 12px 16px 11px 52px;
      `;
    }
    if (hasImage && type === 'radio') {
      return css`
        text-align: center;
        border-radius: 6px;
        padding: 12px;
        display: inline-block;
        width: 48%;
        margin: 1%;
      `;
    }
    if (!hasImage && type === 'checkbox') {
      return css`
        border-radius: 3px;
        padding: 10px 16px 10px 44px;
      `;
    }
    if (hasImage && type === 'checkbox') {
      return css`
        padding: 12px;
        display: inline-block;
        width: 48%;
        margin: 1%;
      `;
    }
    return '';
  }};
  ${({ checked, hasImage }) => {
    if (!hasImage && checked) {
      return css`
        color: ${colore.ui.contrast};
        background-color: ${(props) => props.theme.brand};
        ${media.desktop`
          border-bottom-width: 1px;
        `}
      `;
    }
    if (hasImage && checked) {
      return css`
        box-shadow: 0 0 0 4px ${(props) => props.theme.brand};
        ${media.desktop`
          border: none;
        `}
      `;
    }
    return '';
  }};
  ${({ sbagliato, hasImage }) => {
    if (!hasImage && sbagliato) {
      return css`
        color: ${colore.ui.contrast};
        background-color: ${colore.actions.error};
        ${media.desktop`
          border-bottom-width: 1px;
        `}
      `;
    }
    if (hasImage && sbagliato) {
      return css`
        box-shadow: 0 0 0 4px ${colore.actions.error};
      `;
    }
    return '';
  }};
  ${({ giusto, hasImage }) => {
    if (!hasImage && giusto) {
      return css`
        color: ${colore.ui.contrast};
        background-color: ${colore.actions.okay};
        ${media.desktop`
          border-bottom-width: 1px;
        `}
      `;
    }
    if (hasImage && giusto) {
      return css`
        box-shadow: 0 0 0 4px ${colore.actions.okay};
      `;
    }
    return '';
  }};
  a {
    color: currentColor;
  }
  img {
    max-height: 200px;
  }
`;

RadioCheckLabel.defaultProps = {
  theme: {
    brand: '#00BBEF',
  },
};

export const FakeRadioCheck = styled.span`
  display: block;
  position: absolute;
  width: 18px;
  height: 18px;
  top: 50%;
  margin: -9px 0 0 0;
  border: 4px solid ${(props) => props.theme.brand};
  background: ${colore.ui.contrast};
  box-shadow: inset 0 0 0 3px ${colore.ui.contrast};
  ${({ type }) => type === 'radio' && css`
    left: 17px;
    border-radius: 100%;
  `};
  ${({ type }) => type === 'checkbox' && css`
    left: 12px;
  `};
  ${({ checked }) => checked && css`
    border: 4px solid ${colore.ui.contrast};
    background-color: ${colore.ui.contrast};
    box-shadow: inset 0 0 0 3px ${(props) => props.theme.brand};
  `};
  ${({ sbagliato }) => sbagliato && css`
    border: 4px solid ${colore.ui.contrast};
    background-color: ${colore.ui.contrast};
    box-shadow: inset 0 0 0 3px ${colore.actions.error};
  `};
  ${({ giusto }) => giusto && css`
    border: 4px solid ${colore.ui.contrast};
    background-color: ${colore.ui.contrast};
    box-shadow: inset 0 0 0 3px ${colore.actions.okay};
  `};
  ${({ hasImage }) => hasImage && css`
    display: none;
  `};
`;

FakeRadioCheck.defaultProps = {
  theme: {
    brand: '#00BBEF',
  },
};

class RadioCheck extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    const { label, type, checked, sbagliato, giusto, hasImage } = this.props;
    return (
      <RadioCheckLabel
        type={type}
        checked={checked}
        sbagliato={sbagliato}
        giusto={giusto}
        hasImage={hasImage}
        onClick={this.props.onClickFunction}
      >
        <FakeRadioCheck
          type={type}
          checked={checked}
          readOnly={this.props.readonly}
          sbagliato={sbagliato}
          giusto={giusto}
          hasImage={hasImage}
        />
        {label}
      </RadioCheckLabel>
    );
  }
}

RadioCheck.propTypes = {
  label: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array,
  ]),
  type: PropTypes.string.isRequired,
  checked: PropTypes.bool,
  sbagliato: PropTypes.bool,
  giusto: PropTypes.bool,
  readonly: PropTypes.bool,
  onClickFunction: PropTypes.func,
  hasImage: PropTypes.bool,
};

export default RadioCheck;
