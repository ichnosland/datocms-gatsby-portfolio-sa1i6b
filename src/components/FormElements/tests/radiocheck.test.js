import React from 'react';
import { shallow } from 'enzyme';

import RadioCheck,
{
  RadioCheckLabel,
  FakeRadioCheck,
} from '../RadioCheck';

const type = 'radio';
const checkbox = 'checkbox';
const radio = 'radio';
const theme = {
  brand: '#111111',
};

describe('<RadioCheck />', () => {
  it('should render a RadioCheckLabel', () => {
    const renderedComponent = shallow(<RadioCheck type={type} />);
    expect(renderedComponent.find(RadioCheckLabel).length).toEqual(1);
  });

  it('should render its css w/ props type === radio if !hasImage', () => {
    const renderedComponent = shallow(<RadioCheck type={radio} hasImage={false} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props type === checkbox if !hasImage', () => {
    const renderedComponent = shallow(<RadioCheck type={checkbox} hasImage={false} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props type === radio && hasImage', () => {
    const renderedComponent = shallow(<RadioCheck type={radio} hasImage />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props type === checkbox && hasImage', () => {
    const renderedComponent = shallow(<RadioCheck type={checkbox} hasImage />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props checked', () => {
    const renderedComponent = shallow(<RadioCheck type={type} checked />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props giusto', () => {
    const renderedComponent = shallow(<RadioCheck type={type} giusto />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props sbagliato', () => {
    const renderedComponent = shallow(<RadioCheck type={type} sbagliato />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its label w/ props label', () => {
    const label = 'Etichetta';
    const renderedComponent = shallow(<RadioCheck type={type} label={label} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its label w/ props hasImage', () => {
    const renderedComponent = shallow(<RadioCheck type={type} hasImage />);
    expect(renderedComponent).toMatchSnapshot();
  });
});

describe('<RadioCheckLabel />', () => {
  it('should render a <label> tag', () => {
    const renderedComponent = shallow(<RadioCheckLabel />);
    expect(renderedComponent.type()).toEqual('label');
  });

  it('should render its css w/ props type === radio && !hasImage', () => {
    const renderedComponent = shallow(<RadioCheckLabel type={radio} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props type === checkbox && !hasImage', () => {
    const renderedComponent = shallow(<RadioCheckLabel type={checkbox} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props type === radio && hasImage', () => {
    const renderedComponent = shallow(<RadioCheckLabel type={radio} hasImage />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props type === checkbox && hasImage', () => {
    const renderedComponent = shallow(<RadioCheckLabel type={checkbox} hasImage />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props checked && !hasImage', () => {
    const renderedComponent = shallow(<RadioCheckLabel type={type} checked />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props checked && hasImage', () => {
    const renderedComponent = shallow(<RadioCheckLabel type={type} checked hasImage />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props giusto && !hasImage', () => {
    const renderedComponent = shallow(<RadioCheckLabel type={type} giusto />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props sbagliato && !hasImage', () => {
    const renderedComponent = shallow(<RadioCheckLabel type={type} sbagliato />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props giusto && hasImage', () => {
    const renderedComponent = shallow(<RadioCheckLabel type={type} giusto hasImage />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props sbagliato && hasImage', () => {
    const renderedComponent = shallow(<RadioCheckLabel type={type} sbagliato hasImage />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props bgcolor', () => {
    const bgcolor = 'red';
    const renderedComponent = shallow(<RadioCheckLabel type={type} bgcolor={bgcolor} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});

describe('<FakeRadioCheck />', () => {
  it('should render a <span> tag', () => {
    const renderedComponent = shallow(<FakeRadioCheck />);
    expect(renderedComponent.type()).toEqual('span');
  });

  it('should render its css w/ props theme', () => {
    const renderedComponent = shallow(<FakeRadioCheck checked theme={theme} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props type === radio', () => {
    const renderedComponent = shallow(<FakeRadioCheck type={radio} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props type === checkbox', () => {
    const renderedComponent = shallow(<FakeRadioCheck type={checkbox} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props checked', () => {
    const renderedComponent = shallow(<FakeRadioCheck type={type} checked />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props giusto', () => {
    const renderedComponent = shallow(<FakeRadioCheck type={type} giusto />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props sbagliato', () => {
    const renderedComponent = shallow(<FakeRadioCheck type={type} sbagliato />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props hasImage', () => {
    const renderedComponent = shallow(<FakeRadioCheck type={type} hasImage />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should be checked w/ props readOnly', () => {
    const renderedComponent = shallow(<FakeRadioCheck type={type} readonly />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should be checked w/ onChangeFunction', () => {
    const renderedComponent = shallow(<FakeRadioCheck type={type} onChangeFunction={() => { }} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
