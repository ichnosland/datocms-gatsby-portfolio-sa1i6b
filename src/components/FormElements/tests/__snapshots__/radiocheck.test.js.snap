// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FakeRadioCheck /> should be checked w/ onChangeFunction 1`] = `
.c0 {
  display: block;
  position: absolute;
  width: 18px;
  height: 18px;
  top: 50%;
  margin: -9px 0 0 0;
  border: 4px solid #00BBEF;
  background: rgb(255,255,255);
  box-shadow: inset 0 0 0 3px rgb(255,255,255);
  left: 17px;
  border-radius: 100%;
}

<span
  className="c0"
  type="radio"
/>
`;

exports[`<FakeRadioCheck /> should be checked w/ props readOnly 1`] = `
.c0 {
  display: block;
  position: absolute;
  width: 18px;
  height: 18px;
  top: 50%;
  margin: -9px 0 0 0;
  border: 4px solid #00BBEF;
  background: rgb(255,255,255);
  box-shadow: inset 0 0 0 3px rgb(255,255,255);
  left: 17px;
  border-radius: 100%;
}

<span
  className="c0"
  type="radio"
/>
`;

exports[`<FakeRadioCheck /> should render its css w/ props checked 1`] = `
.c0 {
  display: block;
  position: absolute;
  width: 18px;
  height: 18px;
  top: 50%;
  margin: -9px 0 0 0;
  border: 4px solid #00BBEF;
  background: rgb(255,255,255);
  box-shadow: inset 0 0 0 3px rgb(255,255,255);
  left: 17px;
  border-radius: 100%;
  border: 4px solid rgb(255,255,255);
  background-color: rgb(255,255,255);
  box-shadow: inset 0 0 0 3px #00BBEF;
}

<span
  checked={true}
  className="c0"
  type="radio"
/>
`;

exports[`<FakeRadioCheck /> should render its css w/ props giusto 1`] = `
.c0 {
  display: block;
  position: absolute;
  width: 18px;
  height: 18px;
  top: 50%;
  margin: -9px 0 0 0;
  border: 4px solid #00BBEF;
  background: rgb(255,255,255);
  box-shadow: inset 0 0 0 3px rgb(255,255,255);
  left: 17px;
  border-radius: 100%;
  border: 4px solid rgb(255,255,255);
  background-color: rgb(255,255,255);
  box-shadow: inset 0 0 0 3px rgb(153,226,17);
}

<span
  className="c0"
  type="radio"
/>
`;

exports[`<FakeRadioCheck /> should render its css w/ props hasImage 1`] = `
.c0 {
  display: block;
  position: absolute;
  width: 18px;
  height: 18px;
  top: 50%;
  margin: -9px 0 0 0;
  border: 4px solid #00BBEF;
  background: rgb(255,255,255);
  box-shadow: inset 0 0 0 3px rgb(255,255,255);
  left: 17px;
  border-radius: 100%;
  display: none;
}

<span
  className="c0"
  type="radio"
/>
`;

exports[`<FakeRadioCheck /> should render its css w/ props sbagliato 1`] = `
.c0 {
  display: block;
  position: absolute;
  width: 18px;
  height: 18px;
  top: 50%;
  margin: -9px 0 0 0;
  border: 4px solid #00BBEF;
  background: rgb(255,255,255);
  box-shadow: inset 0 0 0 3px rgb(255,255,255);
  left: 17px;
  border-radius: 100%;
  border: 4px solid rgb(255,255,255);
  background-color: rgb(255,255,255);
  box-shadow: inset 0 0 0 3px rgb(255,0,0);
}

<span
  className="c0"
  type="radio"
/>
`;

exports[`<FakeRadioCheck /> should render its css w/ props theme 1`] = `
.c0 {
  display: block;
  position: absolute;
  width: 18px;
  height: 18px;
  top: 50%;
  margin: -9px 0 0 0;
  border: 4px solid #111111;
  background: rgb(255,255,255);
  box-shadow: inset 0 0 0 3px rgb(255,255,255);
  border: 4px solid rgb(255,255,255);
  background-color: rgb(255,255,255);
  box-shadow: inset 0 0 0 3px #111111;
}

<span
  checked={true}
  className="c0"
/>
`;

exports[`<FakeRadioCheck /> should render its css w/ props type === checkbox 1`] = `
.c0 {
  display: block;
  position: absolute;
  width: 18px;
  height: 18px;
  top: 50%;
  margin: -9px 0 0 0;
  border: 4px solid #00BBEF;
  background: rgb(255,255,255);
  box-shadow: inset 0 0 0 3px rgb(255,255,255);
  left: 12px;
}

<span
  className="c0"
  type="checkbox"
/>
`;

exports[`<FakeRadioCheck /> should render its css w/ props type === radio 1`] = `
.c0 {
  display: block;
  position: absolute;
  width: 18px;
  height: 18px;
  top: 50%;
  margin: -9px 0 0 0;
  border: 4px solid #00BBEF;
  background: rgb(255,255,255);
  box-shadow: inset 0 0 0 3px rgb(255,255,255);
  left: 17px;
  border-radius: 100%;
}

<span
  className="c0"
  type="radio"
/>
`;

exports[`<RadioCheck /> should render its css w/ props checked 1`] = `
<RadioCheck__RadioCheckLabel
  checked={true}
  theme={
    Object {
      "brand": "#00BBEF",
    }
  }
  type="radio"
>
  <RadioCheck__FakeRadioCheck
    checked={true}
    theme={
      Object {
        "brand": "#00BBEF",
      }
    }
    type="radio"
  />
</RadioCheck__RadioCheckLabel>
`;

exports[`<RadioCheck /> should render its css w/ props giusto 1`] = `
<RadioCheck__RadioCheckLabel
  giusto={true}
  theme={
    Object {
      "brand": "#00BBEF",
    }
  }
  type="radio"
>
  <RadioCheck__FakeRadioCheck
    giusto={true}
    theme={
      Object {
        "brand": "#00BBEF",
      }
    }
    type="radio"
  />
</RadioCheck__RadioCheckLabel>
`;

exports[`<RadioCheck /> should render its css w/ props sbagliato 1`] = `
<RadioCheck__RadioCheckLabel
  sbagliato={true}
  theme={
    Object {
      "brand": "#00BBEF",
    }
  }
  type="radio"
>
  <RadioCheck__FakeRadioCheck
    sbagliato={true}
    theme={
      Object {
        "brand": "#00BBEF",
      }
    }
    type="radio"
  />
</RadioCheck__RadioCheckLabel>
`;

exports[`<RadioCheck /> should render its css w/ props type === checkbox && hasImage 1`] = `
<RadioCheck__RadioCheckLabel
  hasImage={true}
  theme={
    Object {
      "brand": "#00BBEF",
    }
  }
  type="checkbox"
>
  <RadioCheck__FakeRadioCheck
    hasImage={true}
    theme={
      Object {
        "brand": "#00BBEF",
      }
    }
    type="checkbox"
  />
</RadioCheck__RadioCheckLabel>
`;

exports[`<RadioCheck /> should render its css w/ props type === checkbox if !hasImage 1`] = `
<RadioCheck__RadioCheckLabel
  hasImage={false}
  theme={
    Object {
      "brand": "#00BBEF",
    }
  }
  type="checkbox"
>
  <RadioCheck__FakeRadioCheck
    hasImage={false}
    theme={
      Object {
        "brand": "#00BBEF",
      }
    }
    type="checkbox"
  />
</RadioCheck__RadioCheckLabel>
`;

exports[`<RadioCheck /> should render its css w/ props type === radio && hasImage 1`] = `
<RadioCheck__RadioCheckLabel
  hasImage={true}
  theme={
    Object {
      "brand": "#00BBEF",
    }
  }
  type="radio"
>
  <RadioCheck__FakeRadioCheck
    hasImage={true}
    theme={
      Object {
        "brand": "#00BBEF",
      }
    }
    type="radio"
  />
</RadioCheck__RadioCheckLabel>
`;

exports[`<RadioCheck /> should render its css w/ props type === radio if !hasImage 1`] = `
<RadioCheck__RadioCheckLabel
  hasImage={false}
  theme={
    Object {
      "brand": "#00BBEF",
    }
  }
  type="radio"
>
  <RadioCheck__FakeRadioCheck
    hasImage={false}
    theme={
      Object {
        "brand": "#00BBEF",
      }
    }
    type="radio"
  />
</RadioCheck__RadioCheckLabel>
`;

exports[`<RadioCheck /> should render its label w/ props hasImage 1`] = `
<RadioCheck__RadioCheckLabel
  hasImage={true}
  theme={
    Object {
      "brand": "#00BBEF",
    }
  }
  type="radio"
>
  <RadioCheck__FakeRadioCheck
    hasImage={true}
    theme={
      Object {
        "brand": "#00BBEF",
      }
    }
    type="radio"
  />
</RadioCheck__RadioCheckLabel>
`;

exports[`<RadioCheck /> should render its label w/ props label 1`] = `
<RadioCheck__RadioCheckLabel
  theme={
    Object {
      "brand": "#00BBEF",
    }
  }
  type="radio"
>
  <RadioCheck__FakeRadioCheck
    theme={
      Object {
        "brand": "#00BBEF",
      }
    }
    type="radio"
  />
  Etichetta
</RadioCheck__RadioCheckLabel>
`;

exports[`<RadioCheckLabel /> should render its css w/ props bgcolor 1`] = `
.c0 {
  position: relative;
  display: block;
  width: 100%;
  margin: 10px 0;
  background-color: red;
  border-radius: 100px;
  padding: 12px 16px 11px 52px;
}

.c0 a {
  color: currentColor;
}

.c0 img {
  max-height: 200px;
}

@media (min-width:64.0625em) {
  .c0 {
    border: 1px solid rgba(0,0,0,0.05);
    border-bottom-width: 2px;
  }
}

<label
  className="c0"
  type="radio"
/>
`;

exports[`<RadioCheckLabel /> should render its css w/ props checked && !hasImage 1`] = `
.c0 {
  position: relative;
  display: block;
  width: 100%;
  margin: 10px 0;
  background-color: rgb(255,255,255);
  border-radius: 100px;
  padding: 12px 16px 11px 52px;
  color: rgb(255,255,255);
  background-color: #00BBEF;
}

.c0 a {
  color: currentColor;
}

.c0 img {
  max-height: 200px;
}

@media (min-width:64.0625em) {
  .c0 {
    border: 1px solid rgba(0,0,0,0.05);
    border-bottom-width: 2px;
  }
}

@media (min-width:64.0625em) {
  .c0 {
    border-bottom-width: 1px;
  }
}

<label
  checked={true}
  className="c0"
  type="radio"
/>
`;

exports[`<RadioCheckLabel /> should render its css w/ props checked && hasImage 1`] = `
.c0 {
  position: relative;
  display: block;
  width: 100%;
  margin: 10px 0;
  background-color: rgb(255,255,255);
  text-align: center;
  border-radius: 6px;
  padding: 12px;
  display: inline-block;
  width: 48%;
  margin: 1%;
  box-shadow: 0 0 0 4px #00BBEF;
}

.c0 a {
  color: currentColor;
}

.c0 img {
  max-height: 200px;
}

@media (min-width:64.0625em) {
  .c0 {
    border: 1px solid rgba(0,0,0,0.05);
    border-bottom-width: 2px;
  }
}

@media (min-width:64.0625em) {
  .c0 {
    border: none;
  }
}

<label
  checked={true}
  className="c0"
  type="radio"
/>
`;

exports[`<RadioCheckLabel /> should render its css w/ props giusto && !hasImage 1`] = `
.c0 {
  position: relative;
  display: block;
  width: 100%;
  margin: 10px 0;
  background-color: rgb(255,255,255);
  border-radius: 100px;
  padding: 12px 16px 11px 52px;
  color: rgb(255,255,255);
  background-color: rgb(153,226,17);
}

.c0 a {
  color: currentColor;
}

.c0 img {
  max-height: 200px;
}

@media (min-width:64.0625em) {
  .c0 {
    border: 1px solid rgba(0,0,0,0.05);
    border-bottom-width: 2px;
  }
}

@media (min-width:64.0625em) {
  .c0 {
    border-bottom-width: 1px;
  }
}

<label
  className="c0"
  type="radio"
/>
`;

exports[`<RadioCheckLabel /> should render its css w/ props giusto && hasImage 1`] = `
.c0 {
  position: relative;
  display: block;
  width: 100%;
  margin: 10px 0;
  background-color: rgb(255,255,255);
  text-align: center;
  border-radius: 6px;
  padding: 12px;
  display: inline-block;
  width: 48%;
  margin: 1%;
  box-shadow: 0 0 0 4px rgb(153,226,17);
}

.c0 a {
  color: currentColor;
}

.c0 img {
  max-height: 200px;
}

@media (min-width:64.0625em) {
  .c0 {
    border: 1px solid rgba(0,0,0,0.05);
    border-bottom-width: 2px;
  }
}

<label
  className="c0"
  type="radio"
/>
`;

exports[`<RadioCheckLabel /> should render its css w/ props sbagliato && !hasImage 1`] = `
.c0 {
  position: relative;
  display: block;
  width: 100%;
  margin: 10px 0;
  background-color: rgb(255,255,255);
  border-radius: 100px;
  padding: 12px 16px 11px 52px;
  color: rgb(255,255,255);
  background-color: rgb(255,0,0);
}

.c0 a {
  color: currentColor;
}

.c0 img {
  max-height: 200px;
}

@media (min-width:64.0625em) {
  .c0 {
    border: 1px solid rgba(0,0,0,0.05);
    border-bottom-width: 2px;
  }
}

@media (min-width:64.0625em) {
  .c0 {
    border-bottom-width: 1px;
  }
}

<label
  className="c0"
  type="radio"
/>
`;

exports[`<RadioCheckLabel /> should render its css w/ props sbagliato && hasImage 1`] = `
.c0 {
  position: relative;
  display: block;
  width: 100%;
  margin: 10px 0;
  background-color: rgb(255,255,255);
  text-align: center;
  border-radius: 6px;
  padding: 12px;
  display: inline-block;
  width: 48%;
  margin: 1%;
  box-shadow: 0 0 0 4px rgb(255,0,0);
}

.c0 a {
  color: currentColor;
}

.c0 img {
  max-height: 200px;
}

@media (min-width:64.0625em) {
  .c0 {
    border: 1px solid rgba(0,0,0,0.05);
    border-bottom-width: 2px;
  }
}

<label
  className="c0"
  type="radio"
/>
`;

exports[`<RadioCheckLabel /> should render its css w/ props type === checkbox && !hasImage 1`] = `
.c0 {
  position: relative;
  display: block;
  width: 100%;
  margin: 10px 0;
  background-color: rgb(255,255,255);
  border-radius: 3px;
  padding: 10px 16px 10px 44px;
}

.c0 a {
  color: currentColor;
}

.c0 img {
  max-height: 200px;
}

@media (min-width:64.0625em) {
  .c0 {
    border: 1px solid rgba(0,0,0,0.05);
    border-bottom-width: 2px;
  }
}

<label
  className="c0"
  type="checkbox"
/>
`;

exports[`<RadioCheckLabel /> should render its css w/ props type === checkbox && hasImage 1`] = `
.c0 {
  position: relative;
  display: block;
  width: 100%;
  margin: 10px 0;
  background-color: rgb(255,255,255);
  padding: 12px;
  display: inline-block;
  width: 48%;
  margin: 1%;
}

.c0 a {
  color: currentColor;
}

.c0 img {
  max-height: 200px;
}

@media (min-width:64.0625em) {
  .c0 {
    border: 1px solid rgba(0,0,0,0.05);
    border-bottom-width: 2px;
  }
}

<label
  className="c0"
  type="checkbox"
/>
`;

exports[`<RadioCheckLabel /> should render its css w/ props type === radio && !hasImage 1`] = `
.c0 {
  position: relative;
  display: block;
  width: 100%;
  margin: 10px 0;
  background-color: rgb(255,255,255);
  border-radius: 100px;
  padding: 12px 16px 11px 52px;
}

.c0 a {
  color: currentColor;
}

.c0 img {
  max-height: 200px;
}

@media (min-width:64.0625em) {
  .c0 {
    border: 1px solid rgba(0,0,0,0.05);
    border-bottom-width: 2px;
  }
}

<label
  className="c0"
  type="radio"
/>
`;

exports[`<RadioCheckLabel /> should render its css w/ props type === radio && hasImage 1`] = `
.c0 {
  position: relative;
  display: block;
  width: 100%;
  margin: 10px 0;
  background-color: rgb(255,255,255);
  text-align: center;
  border-radius: 6px;
  padding: 12px;
  display: inline-block;
  width: 48%;
  margin: 1%;
}

.c0 a {
  color: currentColor;
}

.c0 img {
  max-height: 200px;
}

@media (min-width:64.0625em) {
  .c0 {
    border: 1px solid rgba(0,0,0,0.05);
    border-bottom-width: 2px;
  }
}

<label
  className="c0"
  type="radio"
/>
`;
